bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java

```diff
@@ -600,18 +638,21 @@ static class DestinationTableBuilder implements Callable<TableInfo> {
     final TableId destinationTable;
     final int expirationTimeInMinutes;
     final JobConfigurationFactory jobConfigurationFactory;
+    final Map<String, String> queryJobLabels;
 
     DestinationTableBuilder(
         BigQueryClient bigQueryClient,
         String querySql,
         TableId destinationTable,
         int expirationTimeInMinutes,
-        JobConfigurationFactory jobConfigurationFactory) {
+        JobConfigurationFactory jobConfigurationFactory,
+        Map<String, String> queryJobLabels) {
```

Comment:
JobConfigurationFactory already has a list of labels, can we add the new labels there? If not, let's rename (all over this PR) the parameter name to `additionalQueryJobLabels`

Reply:
Not able to add the new labels there since it is initialized as part of the Guice initialization before we know whether query pushdown will be used or not. I will change the parameter name to `additionalQueryJobLabels`